packet_test.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"net"
+	"reflect"
+	"testing"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+func createMockmDNSPacket(isIPv4 bool, isDNSQuery bool) []byte {
+	var options gopacket.SerializeOptions
+	var ethernetLayer, dot1QLayer, ipLayer, udpLayer, dnsLayer gopacket.SerializableLayer
+
+	ethernetLayer = &layers.Ethernet{
+		SrcMAC:       net.HardwareAddr{0xFF, 0xAA, 0xFA, 0xAA, 0xFF, 0xAA},
```

Comment:
Let's split mock values to constants outside the function scope, this way we can easily reuse them: `var srcMACTest` for example, etc.